starknet_py/net/models/transaction.py

```diff
@@ -94,10 +94,10 @@ class _DeprecatedAccountTransaction(AccountTransaction, ABC):
 @dataclass(frozen=True)
 class _AccountTransactionV3(AccountTransaction, ABC):
     resource_bounds: ResourceBoundsMapping
-    tip: int = field(init=False, default=0)
     nonce_data_availability_mode: DAMode = field(init=False, default=DAMode.L1)
     fee_data_availability_mode: DAMode = field(init=False, default=DAMode.L1)
     paymaster_data: List[int] = field(init=False, default_factory=list)
+    tip: int
```

Comment:
It would make sense to make a check that tip is > 0, can we add it?

Reply:
As follow up, we can add a test with negative tip.

Reply:
This should be enforced by marshmallow, it is typed a u64 there.

I've checked, if you pass a negative tip we currently fail when computing the transaction hash to be signed, we get en error from poseidon.py ` ValueError: -123456 out of u_int256_t range` (for too large non-negative value, marshmallow checks will catch it with `marshmallow.exceptions.ValidationError: Invalid value provided for Uint64: 73786976294838206464`).

Not sure if we should have any explicit checks for the `tip` value, we don't check `nonce` provided by the user or `ResourceBounds`.

Reply:
Hm ok let's leave it like that then.